### PR TITLE
Add same receiver name tenet

### PR DESCRIPTION
**Pull request by leilaes**

### CODE REVIEW

File: tenets/codelingo/code-review-comments/same-receiver-name/codelingo.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    name as typeName
+          go.ident:
+            name as methodName1
+        go.func_decl(depth = any):       
+          go.field_list:
+            go.field:
+              go.names:
+                go.ident:
+                  name as receiverName2
+                  notEqual(receiverName1, receiverName2)
```
Comment (Daanikus) [inline tenets/codelingo/code-review-comments/same-receiver-name/codelingo.yaml]: Have you tried `receiverName1 != receiverName2`? I think it's supported

File: tenets/codelingo/code-review-comments/same-receiver-name/test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import "fmt"
+
+type T1 struct{}
+type T2 struct{}
+
+func (t T1) Method1() {
+	fmt.Println("Hello from Method1")
+}
+
+func (this T1) Method2() {
```
Comment (Daanikus) [inline tenets/codelingo/code-review-comments/same-receiver-name/test.go]:
```suggestion
func (this T1) Method2() { // ISSUE
```

File: tenets/codelingo/code-review-comments/same-receiver-name/codelingo.yaml (hunk 1 above)
Comment (leilaes) [reply tenets/codelingo/code-review-comments/same-receiver-name/codelingo.yaml]: Unfortunately, it doesn't work.

File: tenets/codelingo/code-review-comments/same-receiver-name/test.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package main
+
+import "fmt"
+
+type T1 struct{}
+type T2 struct{}
+
+func (t T1) Method1() { //Issue
```
Comment (BlakeMScurr) [inline tenets/codelingo/code-review-comments/same-receiver-name/test.go]:
```suggestion
func (t T1) Method1() { //ISSUE
```

I like to yell it 😛

File: tenets/codelingo/code-review-comments/same-receiver-name/test.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	fmt.Println("Hello from Method1")
+}
+
+func (this T1) Method2() { //Issue
```
Comment (BlakeMScurr) [inline tenets/codelingo/code-review-comments/same-receiver-name/test.go]:
```suggestion
func (this T1) Method2() { //ISSUE
```

**Pull request merged**